app/Policies/ServerPolicy.php

```diff
@@ -40,6 +40,23 @@ public function __construct()
     }
 
     /**
+     * Checks if the user has the given permission on/for the server.
+     *
+     * @param \Pterodactyl\Models\User   $user
+     * @param \Pterodactyl\Models\Server $server
+     * @param $permission
+     * @return bool
+     */
+    private function checkPermission(User $user, Server $server, $permission)
+    {
+        if ($this->isOwner($user, $server)) {
+            return true;
+        }
+
+        return $user->permissions()->server($server)->permission($permission)->exists();
```

Comment:
This works out nicely because we can always change this query in the future. I see what you mean, but this keeps it pretty flexible for now.